src/Pact/Runtime/Capabilities.hs

```diff
@@ -154,31 +147,65 @@ evalUserCapability i af scope cap cdef test = go scope
 
     pushSlot s = evalCapabilities . capStack %= (s:)
 
+defCapMetaParts :: HasInfo i => i -> UserCapability -> DefcapMeta a -> Def Ref -> Eval e (Int, SigCapability, PactValue)
+defCapMetaParts i cap (DefcapMeta _ argName) cdef = case findArg argName of
+  Nothing ->
+    evalError' cdef $ "Invalid managed argument name: " <> pretty argName
+  Just idx -> case decomposeManaged' idx cap of
+    Nothing -> evalError' i $ "Missing argument index from capability: " <> pretty idx
+    Just (static,v) -> return (idx,static,v)
+  where
+    findArg an = foldl' (matchArg an) Nothing (zip [0..] $ _ftArgs $ _dFunType cdef)
+    matchArg _ (Just idx) _ = Just idx
+    matchArg an Nothing (idx,Arg{..}) | _aName == an = Just idx
+                                      | otherwise = Nothing
```

Comment:
We can save a few cycles here: `findArg` is `findIndex an $ _aName <$> _ftArgs (_dFunType cdef)`. Vector has an efficient fusable implementation of this.